src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy

```diff
@@ -24,16 +24,27 @@ class VersionResolver {
     private final ScmRepository repository
 
     private final VersionSorter sorter
+    private ScmPosition latestChangePosition
+    private String projectRootRelativePath
 
     VersionResolver(ScmRepository repository) {
+        this(repository, null)
+    }
+
+    VersionResolver(ScmRepository repository, String projectRootRelativePath) {
         this.repository = repository
+        this.projectRootRelativePath = projectRootRelativePath
         this.sorter = new VersionSorter()
     }
 
     VersionContext resolveVersion(VersionProperties versionRules, TagProperties tagProperties, NextVersionProperties nextVersionProperties) {
-        ScmPosition position = repository.currentPosition()
+        if (projectRootRelativePath != null) {
+            latestChangePosition = repository.positionOfLastChangeIn(projectRootRelativePath);
```

Comment:
Unfortunately I cannot call methods on repository in the constructor of VersionResolver and set "latestChangePosition" there. 
In the tests, the constructor is called in setup(), before any commits exist in the repo.